Fix source maps issue resulting in a crash

RobinMalfait opened the pull request:
This PR fixes an issue where source maps weren't properly working. This is because we started using Lightning CSS in between our PostCSS plugins.

The implementation of Lightning CSS currently re-parses the CSS therefore losing the original information from the `root` tree. After that we reset the result.root based on the result of Lightning CSS.

This PR should resolve the source map issues by passing the previous source maps to Lightning CSS, and passing the resulting source maps from Lightning CSS back to PostCSS.
<!--

👋 Hey, thanks for your interest in contributing to Tailwind!

**Please ask first before starting work on any significant new features.**

It's never a fun experience to have your pull request declined after investing a lot of time and effort into a new feature. To avoid this from happening, we request that contributors create an issue to first discuss any significant new features. This includes things like adding new utilities, creating new at-rules, or adding new component examples to the documentation.

https://github.com/tailwindcss/tailwindcss/blob/master/.github/CONTRIBUTING.md

-->

RobinMalfait committed:
only try to pretty print `err.source` if it exists

RobinMalfait committed:
ensure source maps are generated properly

RobinMalfait committed:
enable all source map tests

RobinMalfait committed:
update annotation values in tests

RobinMalfait committed:
fix small typo

RobinMalfait commented on tests/source-maps.test.js:
```diff
@@ -38,33 +38,34 @@ test.skip('apply generates source maps', async () => {
-    '13:8 -> 22:8',
-    '13:31 -> 23:0',
-  ])
+  expect(annotations).toMatchInlineSnapshot(`
```
Now that we don't use `crosscheck` anymore, we can use `toMatchInlineSnapshot` again!

thecrypticace replied:
🎉🎉🎉

RobinMalfait committed:
drop unnecessary path resolving

RobinMalfait committed:
add `from` and `to` to `run` test helper

RobinMalfait commented on tests/source-maps.test.js:
```diff
@@ -38,33 +38,34 @@ test.skip('apply generates source maps', async () => {
   // And always be able to point to the original source file
   expect(sources).not.toContain('<no source>')
-  expect(sources.length).toBe(1)
+  expect(sources.length).toBe(2)
```
All of these are increased by 1. If you (temporarily) set the `minify: false` to `minify: true` in the lightningCssPlugin then it goes down by 1 again.

thecrypticace replied:
the number of sources should definitely just be `1`

Introducing additional sources that disappear during minification sounds like a lightningcss bug. I'll set up an isolated test case for that and file a bug report.

RobinMalfait committed:
handle `map: true`

RobinMalfait committed:
add test for `map: { inline: true }`

RobinMalfait committed:
force Lightning CSS to handle an inline source map

RobinMalfait committed:
base64 encode the test name

This will get rid of special characters that can cause issues /
confusion when debugging tests.

RobinMalfait committed:
ensure utilities and components with variants are pointing to the correct source

Co-authored-by: Jordan Pittman <jordan@cryptica.me>

RobinMalfait committed:
update changelog

thecrypticace reviewed (approved):
aside from the source path issues which is probs a lightning bug this LGTM. We can merge it and take care of those later.
